apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -6,15 +6,96 @@ import {
   CommandLineStringParameter,
   ICommandLineActionOptions
 } from '@rushstack/ts-command-line';
+import { SyncHook, AsyncParallelHook, AsyncSeriesHook } from 'tapable';
+import { performance } from 'perf_hooks';
 
 import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+import { CleanAction } from './CleanAction';
 
-export interface IBuildActionOptions extends IHeftActionBaseOptions {}
+export interface IBuildActionOptions extends IHeftActionBaseOptions {
+  cleanAction: CleanAction;
+}
+
+/**
+ * @public
+ */
+export class BuildPhaseHooksBase {
+  public readonly run: AsyncParallelHook = new AsyncParallelHook();
+}
+
+/**
+ * @public
+ */
+export interface IBuildPhase<TBuildPhaseHooks extends BuildPhaseHooksBase = BuildPhaseHooksBase> {
+  hooks: TBuildPhaseHooks;
+}
+
+/**
+ * @public
+ */
+export interface ISharedCopyStaticAssetsConfiguration {
+  /**
+   * File extensions that should be copied from the src folder to the destination folder(s)
+   */
+  fileExtensions?: string[];
+
+  /**
+   * Paths or globs that should be explicitly excluded. This takes precedence over paths listed in \"include\".
+   */
+  exclude?: string[];
+
+  /**
+   * Paths or globs that should be explicitly included.
+   */
+  include?: string[];
+}
+
+/**
+ * @public
+ */
+export interface ICopyStaticAssetsConfiguration extends ISharedCopyStaticAssetsConfiguration {
+  /**
+   * The folder from which assets should be copied
```

Comment:
We can save a lot of time by procrastinating documentation until later when the design is solidified and matured a bit.

HOWEVER, I feel like the task config options are the one item that's important to document comprehensively, since it's the spec.  How can we debug the implementation without knowing how it was intended to behave?

Just the basics like:
- Why would I use this setting?
- What are the valid inputs?
- If I omit it, is there a default value?
- Show an example value
- How do the options interact with each other?
- Can it be a relative path? If so, what's it resolved relative to?